integration-libs/digital-payments/src/checkout/cms-components/dp-payment-method/dp-payment-callback/dp-payment-callback.component.spec.ts

```diff
@@ -0,0 +1,246 @@
+        mockSessionId,
+        mockSignature
+      );
+    });
+
+    it('should show payment fetch error when empty payment details', () => {
+      const mockDpPaymentRequest: DpPaymentRequest = {
+        sessionId: mockSessionId,
+        signature: mockSignature,
+      };
+      spyOn(dpStorageService, 'readCardRegistrationState').and.returnValue(
+        mockDpPaymentRequest
+      );
+      spyOn(dpPaymentService, 'createPaymentDetails').and.returnValue(of({}));
+
+      component.ngOnInit();
+
+      expect(dpStorageService.readCardRegistrationState).toHaveBeenCalled();
+      expect(dpPaymentService.createPaymentDetails).toHaveBeenCalledWith(
+        mockSessionId,
+        mockSignature
+      );
+      expect(msgService.add).toHaveBeenCalledWith(
+        { key: 'dpPaymentForm.error.paymentFetch' },
+        GlobalMessageType.MSG_TYPE_ERROR
+      );
+      expect(component.closeCallback.emit).toHaveBeenCalled();
+    });
+
+    it('should show unknown error when empty payment request', () => {
+      spyOn(dpStorageService, 'readCardRegistrationState').and.returnValue({});
+
+      component.ngOnInit();
+
+      expect(dpStorageService.readCardRegistrationState).toHaveBeenCalled();
+      expect(msgService.add).toHaveBeenCalledWith(
+        { key: 'dpPaymentForm.error.unknown' },
+        GlobalMessageType.MSG_TYPE_ERROR
+      );
+      expect(component.closeCallback.emit).toHaveBeenCalled();
+    });
+  });
+});
+
+describe('DpPaymentCallbackComponent without query param', () => {
+  let component: DpPaymentCallbackComponent;
+  let fixture: ComponentFixture<DpPaymentCallbackComponent>;
+  let msgService: GlobalMessageService;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [RouterModule.forRoot([]), StoreModule.forRoot({})],
+      declarations: [
+        DpPaymentCallbackComponent,
+        MockTranslatePipe,
+        MockSpinnerComponent,
+      ],
+      providers: [
+        {
+          provide: DpPaymentCallbackComponent,
+          useClass: DpPaymentCallbackComponent,
+        },
+        {
+          provide: ActivatedRoute,
+          useValue: emptyActivatedRoute,
+        },
+        {
+          provide: DpCheckoutPaymentService,
+          useClass: MockDpCheckoutPaymentService,
+        },
+        {
+          provide: DpLocalStorageService,
+          useClass: MockDpLocalStorageService,
+        },
+        {
+          provide: GlobalMessageService,
+          useClass: GlobalMessageService,
+        },
+      ],
+    }).compileComponents();
+
+    msgService = TestBed.inject(GlobalMessageService);
+
+    spyOn(msgService, 'add').and.stub();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(DpPaymentCallbackComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('ngOnInit()', () => {
```

Comment:
Can we have a unit test for the successful response?

Reply:
done